### PR TITLE
Homework_Week2_Cemal

### PR DESCRIPTION
**Pull request by CmlSph**

### CODE REVIEW

File: week2/project/hackyourweather/src/Search.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <div className="app_container">
+      <form>
+        <input
+          className="searchInput"
+          type="text"
+          placeholder="Enter the City Name ...."
+          onChange={getName}
+          value={cityName}
+        />
+      </form>
```
Comment (Meno-101) [inline week2/project/hackyourweather/src/Search.js:21]: the `form` element here may have a certain use, in which you didn't use.

to explain, Forms are used to contain certain elements and tags in order to have a specific behavior and trigger particular events.

in your case, when I fill the input in the form and press enter the page reloads, that is because of the default behavior of the onSubmit event, clicking enter or having a submit button are the same thing.

now to explain the actual problem we need to go back to HTML class, can you describe how to order an html form elements?

please don't get me wrong but when somebody has a problem of understanding one of the things he gotta get back to basics, if you knew how to order the html form elements, then then it's probably js2 or js3 where you learned about forms.

In short what you need, is a look back on how forms work (in general) because you don't have much time, and then comeback here and do a certain change where you will either have a submit button with a prevented default behavior or moving the button you have in your program next to the input "inside the form", or something else I will not say it, because I want you to learn from your mistakes.

**Review by Meno-101 — CHANGES_REQUESTED** [review]
It's working, but I would like to have some more solid understanding of forms and their default behavior, will be approved on the right changes regarding the form element in the application.

anyother comments will be after the changes applied.

**Comment by CmlSph** [comment]
@Meno-101 thanks for the feedback. Can you please take a look at it again?

File: week2/project/hackyourweather/src/Search.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import "./style.css";
+import WeatherInfo from "./WeatherInfo";
+
+const API_KEY = process.env.REACT_APP_OPENWEATHERMAP_API_KEY;
+console.log(API_KEY);
```
Comment (Meno-101) [inline week2/project/hackyourweather/src/Search.js:6]: And don't leave something like this in production, especially when you have a key, this is serious matter.

An API key is just like a password, once someone have an api key of someone else is just like having access to his account blindly on that website.

Same thing applies for the `.env` file but it's just an exercise that's why I didn't yet comment about it.

**Review by Meno-101 — COMMENTED** [review]
LGTM; I will not request any changes since there are no errors.

the only thing that I will leave to you is having a look at the comments below and considering what can be improved.

I will approve after the changes.
